snuba/migrations/snuba_migrations/transactions/0015_transactions_add_source_column.py

```diff
@@ -0,0 +1,52 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Column, String
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    """
+    The source column is required to query for transaction events
+    in the transaction summary when the transaction name has been
```

Comment:
How is this going to be used? In every query on the transactions summary page?

Reply:
@jan-auer  can you please help here to answer?

Reply:
This will be used to render a special version of the transaction summary page where we don't query transaction events by transaction name, but rather by source. It is to collect all the transaction events where we dropped the transaction name during metric extraction.

The majority of transaction summary pages will still use the regular transaction names, so this is for the cases where we get unsanitized URLs from browsers, and such. 

cc @k-fish for details.

Reply:
Let me give more context: I ask because if you are using this column to filter every/a lot of queries on the performance page, then it might be worth also adding an index of some kind. But this is also a "potential future problem" more than a "fix in this PR" problem.

Reply:
@evanh do you want me to add TODO or create a ticket to not forget about it?

Reply:
Just my 2c - Based on the comment below this is an extremely low cardinality column with only 7 unique values. Low cardinality column type should already be fairly optimized and I don't know that an index here would really improve much (Clickhouse only has data skipping indexes). I wouldn't worry about it unless it's a problem.

Reply:
I don't think there's any further action here. Let's see how it works and if there's a performance issue we can revisit it.

Reply:
For usage context;

Upon LA we'll be either querying this across all transactions on transaction summary, or potentially directing people to a discover page to list their 'high cardinality' transactions that need to be manually fixed. We aren't planning on querying this by default on any views, only for the person who is wondering what 'unparameterized' means on their metrics data and clicks something to look at it.